source/common/network/addr_family_aware_socket_option_impl.cc

```diff
@@ -29,7 +30,7 @@ absl::optional<Address::IpVersion> getVersionFromSocket(const Socket& socket) {
     if (socket.localAddress()) {
       return {getVersionFromAddress(socket.localAddress())};
     } else {
-      return {getVersionFromAddress(Address::addressFromFd(socket.ioHandle().fd()))};
+      return {getVersionFromAddress(SocketInterface::addressFromFd(socket.ioHandle().fd()))};
```

Comment:
Do you want to fix this fd indirection in this PR or a follow up?

Reply:
I was planning on doing it in a subsequent PR (after the last in this series). Obviously, if that's fine with you :-)

Reply:
Yes that's fine.